### PR TITLE
feat: modernisasi UI peta dan tambah dukungan wilayah sampai desa

### PR DESCRIPTION
**Pull request by dendyadinirwana**
## Ringkasan

PR ini memperbarui tampilan dan interaksi aplikasi peta wilayah agar lebih modern, map-first, dan lebih nyaman digunakan untuk eksplorasi data administrasi Indonesia.

Perubahan utama:
- membuat peta tampil full-height sampai bawah viewport
- mengubah panel filter wilayah menjadi floating sidebar yang bisa di-collapse
- menambahkan filter wilayah sampai level kecamatan dan desa/kelurahan
- menampilkan metadata wilayah (`kode`, `lat`, `lng`) langsung di dalam peta
- menambahkan fitur klik pada peta untuk memilih wilayah terdekat
- memperbaiki kesadaran tema dark/light pada tile peta dan kontrol Leaflet
- memindahkan informasi versi, copyright, dan source ke navbar atas

## Latar Belakang

Sebelumnya aplikasi lebih berfokus pada pemilihan wilayah dasar melalui dropdown dan visualisasi peta sederhana. Dalam pengembangan ini, aplikasi diarahkan agar lebih nyaman digunakan sebagai antarmuka eksplorasi wilayah berbasis peta.

Selain itu, terdapat kebutuhan untuk mulai mendukung data batas wilayah sampai level kecamatan dan desa/kelurahan.

Namun, dalam proses integrasi data SHP, ditemukan bahwa nama kecamatan dan desa/kelurahan dari resource geospasial tidak selalu 100% sama dengan nama wilayah yang tersedia di tabel referensi administrasi. Perbedaan ini dapat terjadi karena beberapa hal:

- perbedaan penulisan nama wilayah
- penggunaan spasi yang berbeda, contoh `Margasari` vs `Marga Sari`
- perbedaan kapitalisasi
- perbedaan format administratif antara sumber geospasial dan referensi kode wilayah
- kemungkinan pemekaran atau pembaruan nama wilayah pada salah satu sumber
- adanya polygon/path yang secara geometri tidak selalu sesuai dengan centroid wilayah

Karena itu, perubahan pada PR ini juga menambahkan beberapa mekanisme defensif agar aplikasi tetap dapat menampilkan lokasi secara masuk akal meskipun data geometri belum sepenuhnya bersih.

### 1. Tampilan peta full-height

Peta sekarang mengisi area viewport di bawah navbar, sehingga tidak ada lagi area kosong besar di bawah peta.

Perubahan:
- menghapus padding utama di sekitar peta
- membuat tinggi peta mengikuti `calc(100vh - 52px)`
- menyembunyikan footer bawah agar peta bisa memenuhi layar
- memindahkan informasi versi/copyright/source ke navbar

### 2. Floating sidebar filter wilayah

Panel filter wilayah dipindahkan menjadi floating sidebar di atas peta.

Perubahan:
- panel filter berada di kiri atas peta
- panel bisa di-collapse
- status collapse disimpan di `localStorage`
- ikon collapse dirapikan agar lebih presisi secara visual
- saat panel dibuka/tutup, Leaflet map di-refresh dengan `invalidateSize()`

### 3. Dropdown sampai level kecamatan dan desa/kelurahan

Filter wilayah sekarang mendukung struktur bertingkat:

- Provinsi
- Kota/Kabupaten
- Kecamatan
- Desa/Kelurahan

Dropdown anak dimuat secara bertahap berdasarkan kode parent.

### 4. Klik peta untuk memilih wilayah

Ditambahkan endpoint reverse lookup:

`apps/inc/reverse_lookup.php`

Fitur ini memungkinkan user klik pada peta, lalu aplikasi mencoba menentukan wilayah terdekat berdasarkan koordinat klik.

Alur:
- user klik peta
- sistem membaca `lat/lng`
- backend mencari wilayah kandidat terdekat
- frontend mengisi dropdown provinsi, kota/kabupaten, kecamatan, dan desa/kelurahan jika tersedia
- peta diarahkan ke wilayah hasil pencarian

### 5. Metadata wilayah di dalam peta

Informasi berikut dipindahkan ke overlay compact di kiri bawah peta:

- Kode
- Lat
- Lng

### 6. Kontrol zoom dan basemap

Perubahan:
- tombol zoom `+` dan `-` dipindahkan ke kanan bawah
- kontrol basemap tetap di kanan atas
- styling kontrol Leaflet disesuaikan dengan token tema aplikasi

### 7. Dark/light theme awareness

Peta sekarang lebih sadar tema:

- light theme menggunakan OpenStreetMap terang
- dark theme menggunakan CARTO dark tile
- overlay label satelit ikut menyesuaikan theme
- warna kontrol Leaflet ikut token dark/light theme

## Catatan terkait data kecamatan dan desa

PR ini mulai mendukung batas kecamatan dan desa/kelurahan, tetapi data dari resource geospasial belum sepenuhnya cocok dengan tabel referensi wilayah.

Karena itu, beberapa penyesuaian dilakukan:

- dropdown tetap mengikuti struktur kode wilayah resmi yang tersedia di database
- data geometri digunakan jika cocok dan masuk akal terhadap centroid
- jika polygon/path terdeteksi menyimpang jauh, aplikasi memakai fallback geometry
- reverse lookup mengutamakan hasil yang paling dekat dan paling masuk akal secara lokasi

Hal ini dilakukan agar aplikasi tetap stabil sambil membuka ruang untuk proses lanjutan berupa normalisasi data SHP dan pencocokan nama wilayah secara lebih sistematis.

## File yang Diubah

Perubahan utama:
- `apps/index.php`
- `apps/inc/geo_js.php`
- `apps/inc/geo_ajax.php`
- `apps/js/wilayah.js`
- `apps/inc/reverse_lookup.php`
- `apps/inc/change.color.php`

## Verifikasi

Sudah dicek secara lokal:

```text
php -l apps/index.php
php -l apps/inc/geo_js.php
php -l apps/inc/geo_ajax.php
php -l apps/inc/reverse_lookup.php
```

Manual browser verification:
- peta tampil sampai bawah viewport
- sidebar filter bisa collapse/expand
- dropdown wilayah bertingkat berjalan
- klik peta mengubah filter wilayah
- metadata kode/lat/lng tampil di kiri bawah peta
- tombol zoom berada di kanan bawah
- dark/light theme mengganti warna peta dan kontrol

### CODE REVIEW

**Comment by dendyadinirwana** [comment]
Sebelumnya terima kasih sudah selalu update untuk repo ini. Izin mas @cahyadsn ini PR jika dibutuhkan, cukup banyak perubahan pada UI dan bantu sampai level Desa.

**Comment by cahyadsn** [comment]
terima kasih penambahan fiturnya mas @dendyadinirwana  &#128077;
